salt/utils/event.py

```diff
@@ -430,18 +468,31 @@ def get_event(self, wait=5, tag='', full=False, use_pending=False,
              - 'regex' : regex search '^' + tag event tags
             Default is opts['event_match_type'] or 'startswith'
 
```

Comment:
Ah

This breaks the caching. If `_get_event` is called with a different `match_type` than other areas of code are using, the `pending_events` doesn't collect events properly. I guess the issue would only be problematic if the code using `match_type` is running along with other code though.

Fix is in `_check_pending` and `_get_event` when checking for events to cache (based on what was `subscribe`d to), we just need to ignore the match_type and default to startswith. This will **change behaviour** but only where there is code that is trying to use the removed `pending_tags` functionality as well as `match_type`. If there isn't any code using both then it should be fine I think.

If there is other code using `pending_tags` and also `match_type` a larger fix would be needed. First is what is already done which is move from the `pending_tags` way to the `subscribe` way, and change the `subscribe` and `_check_pending` algorithms to allow a custom match_type.